tools/windows-node-installer/pkg/client/openshift.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	"fmt"
+
+	v1 "github.com/openshift/api/config/v1"
+	clientset "github.com/openshift/client-go/config/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	logger "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// log is the global logger for the client package.
+// Each log record produced by this logger will have an identifier containing `client` tag.
+var log = logger.Log.WithName("client")
+
+// OpenShift is an Client struct which will be used for all OpenShift related functions to interact with the existing
+// Cluster.
+type OpenShift struct {
+	Client *clientset.Clientset
+}
+
+// GetOpenShift uses kubeconfig to create a client for existing OpenShift cluster and returns it or an error.
+func GetOpenShift(kubeConfigPath string) (*OpenShift, error) {
+	log.Info("kubeconfig source: ", kubeConfigPath)
+	rc, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	oc, err := clientset.NewForConfig(rc)
+	if err != nil {
+		return nil, err
+	}
+	return &OpenShift{oc}, nil
+}
+
+// GetCloudProvider returns the Provider details of a given OpenShift client including provider type and region or
+// an error.
+func (o *OpenShift) GetCloudProvider() (*v1.PlatformStatus, error) {
+	infra, err := o.getInfrastructure()
+	if err != nil {
+		return nil, err
+	}
+	if infra.Status == (v1.InfrastructureStatus{}) || *infra.Status.PlatformStatus == (v1.PlatformStatus{}) {
```

Comment:
`infra.Status.PlatformStatus ` Shouldn't this be nil check ?

Reply:
I'm not understanding this. Isn't `*infra.Status.PlatformStatus == (v1.PlatformStatus{})` checking for nil already?